plugins/inputs/opcua/opcua_client.go

```diff
@@ -0,0 +1,432 @@
+package opcua_client
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"time"
+
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/ua"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// OpcUA type
+type OpcUA struct {
+	Name           string   `toml:"name"`
+	Endpoint       string   `toml:"endpoint"`
+	SecurityPolicy string   `toml:"security_policy"`
+	SecurityMode   string   `toml:"security_mode"`
+	Certificate    string   `toml:"certificate"`
+	PrivateKey     string   `toml:"private_key"`
+	Username       string   `toml:"username"`
+	Password       string   `toml:"password"`
+	AuthMethod     string   `toml:"auth_method"`
+	Interval       string   `toml:"time_interval"`
+	TimeOut        int      `toml:"timeout"`
```

Comment:
should both be a `config.Duration`?

Reply:
There may have been a solid reason for interval using a string. I'll look into it and change as appropriate.

Reply:
Did you look into this?  It seems error prone to allow people to set the interval as a string.  The docs say 5s or 10s which is the format config.Duration uses.